pkgs/applications/science/programming/fdr/default.nix

```diff
@@ -0,0 +1,87 @@
+{stdenv, fetchurl, qt57, ncurses, xorg, zlib, python27Packages}:
+let libs = [
+  stdenv.cc.cc
+  python27Packages.python
+  qt57.qtbase
+  qt57.qtx11extras
+  ncurses
+  xorg.libX11
+  xorg.libXft
+  zlib
+  ];
+in
+stdenv.mkDerivation {
+  name = "fdr";
+  src = fetchurl {
+    url = https://www.cs.ox.ac.uk/projects/fdr/downloads/fdr-3754-linux-x86_64.tar.gz;
+    md5 = "c09a05a5d9b54fc65ed12de8b7cdfc11";
+  };
+
+  phases = [ "unpackPhase" "installPhase" ];
+  installPhase = ''
+    mkdir -p "$out"
+
+    # shipped Qt is buggy
+    rm lib/libQt*
+    rm -r lib/qt_plugins
+
+    cp -r * "$out"
+    # Hack around lack of libtinfo in NixOS
+    ln -s ${ncurses.out}/lib/libncursesw.so.6 $out/lib/libtinfo.so.5
+    ln -s ${qt57.qtbase.out}/lib/qt5/plugins $out/lib/qt_plugins
+    ln -s ${zlib.out}/lib/libz.so.1 $out/lib/libz.so.1
```

Comment:
The ncurses one is because the binaries are linked to libtinfo, the qt_plugins one because the way the shared objects are loaded seems to be nonstandard, and this was the only way I could get it to work. I don't know why the zlib one is necessary, but I get `ldd` not found errors without it for some reason...